js/script.js

```diff
@@ -0,0 +1,211 @@
+    userScore: 0,
+    computerScore: 0,  
+    lastWinner: "",
+    userChoice: "",
+    computerChoice: "",
+    progress: [],
+}
+
+initialize ();
+
+
+function initialize () {
+    disableButtons(true);
+    registerListeners();
+}
+
+function registerListeners() {
+    reset.addEventListener('click', function() {
+        state.userScore = 0;
+        state.computerScore = 0;
+        state.gameRounds++;
+        disableButtons(false);
+        roundsLimit();  
+        output.classList.remove('hide');
+        document.getElementById('result').classList.remove('hide');
+        document.getElementById('limit').classList.remove('hide');
+        output.innerHTML = "";
+        result.innerHTML = "";
+    });
+    for (var i = 0; i < buttons.length; i++) { 
+        buttons[i].addEventListener("click", function () {
+            state.userChoice = this.getAttribute("data-move"); 
+            startSingleRound(state.userChoice);
+        });
+    }
+};
+
+function disableButtons(state) {
+    rock.disabled = state;
+    paper.disabled = state;   
+    scissors.disabled = state;    
+};
+
+function roundsLimit() {
+     
+    state.rounds = parseInt(window.prompt('How many round would you play?'));
+    if (state.rounds > 0) {
+       limit.innerHTML = 'We will play up to ' + state.rounds + ' wins';
+    }
+    else if (isNaN(state.rounds) || state.rounds <= 0) {
+       limit.innerHTML = 'Please type correct number of rounds';
+    }
+    return state.rounds ;
+};
+
+function computerChoice() {
+    var compChoice = Math.floor(Math.random() * 3 + 1);
+    if (compChoice === 1) {
+        compChoice = 'rock';
+    } 
+    else if (compChoice === 2) {
+        compChoice = 'paper';    
+    } 
+    else {
+        compChoice = 'scissors';
+    }
+    state.computerChoice = compChoice;
+    return compChoice;
+};
+
+
+
+function compare(userChoice, compChoice) {
+    if (userChoice === compChoice) {
+        state.roundWinner = 'NO ONE WINS this round:';
+        state.lastWinner = "it \'s tie";
+    }
+    else if ((userChoice === "rock" && compChoice === "scissors") || 
+        (userChoice === "paper" && compChoice === "rock")||
+        (userChoice === "scissors" && compChoice === "paper")) {
+        state.roundWinner = 'YOU WON:';
+        state.userScore++;
+        state.lastWinner = "player";          
+    } 
+    else {
+        state.roundWinner = 'COMPUTER WON this round:';
+        state.computerScore++;
+        state.lastWinner = "computer";
+        
+    }
+    return state.roundWinner ;   
+};
+
+
+function clearBox(idElement) {
+    idElement.innerHTML= "";
+};
+
+function roundResultDisplay(roundWinner) {
+    output.insertAdjacentHTML('afterbegin', roundWinner);
+    
+};
+
+function setScore() {
+    document.getElementById('result').innerHTML = (state.userScore + ' - ' + state.computerScore);   
+};
+
+
+function gameIsOver() {
+    if (state.userScore === state.rounds) {  
+        ShowHideModalContent()     
```

Comment:
1) nie nazywamy funkcji dużą literą (to ważne)
2) ten `innerHTML`, który nam został - też powinien być częścią tej funkcji, rozumiem dlaczego go zostawiłaś (bo tak było łatwiej :P) ale to musi wylądować w tej nowej funkcji -> przekaż  tej funkcji jeden argument, który będzie odpowiedzialny za ustalanie `innerHTML` wewnątrz tej funkcji